src/pl/senla/hotel/comparators/RoomReservationsComparatorByGuestName.java

```diff
@@ -1,18 +1,19 @@
 package pl.senla.hotel.comparators;
 
+import pl.senla.hotel.application.annotation.GetInstance;
 import pl.senla.hotel.entity.Guest;
 import pl.senla.hotel.entity.services.RoomReservation;
 import pl.senla.hotel.repository.Repository;
-import pl.senla.hotel.repository.RepositoryGuestCollection;
 
 import java.util.Comparator;
 
 public class RoomReservationsComparatorByGuestName implements Comparator<RoomReservation> {
 
+    @GetInstance(beanName = "RepositoryGuestCollection")
     private final Repository<Guest> repositoryGuest;
 
-    public RoomReservationsComparatorByGuestName() {
-        this.repositoryGuest = RepositoryGuestCollection.getRepositoryGuest();
+    public RoomReservationsComparatorByGuestName(Repository<Guest> repositoryGuest) {
```

Comment:
why do you need a repo here?? it's a casual comparator, let him compare objects without calling repo

Reply:
because task 4.3 was  create "List of guests and their rooms (sort alphabetically)".
Entity RoomReservation has only Guest's id.
Guest's name we can receive only from entity Guest by id from GuestRepository.